src/components/notifications/AddNotification.tsx

```diff
@@ -1,365 +1,476 @@
-                state.view = ViewType.LOADING;
+                state.view = ViewType.LOADING
                 this.setState(state, () => {
-                    getPipelines(state.appliedFilters).then((response) => {
-                        let selectedPipelines = this.state.pipelineList.filter(pipeline => pipeline.checkbox.isChecked);
-                        let newPipelines = response.result || [];
-                        this.setState({ view: ViewType.FORM, pipelineList: selectedPipelines.concat(newPipelines), openSelectPipeline: false });
-                    }).catch((error) => {
-                        showError(error);
-                    })
-                });
+                    getPipelines(state.appliedFilters)
+                        .then((response) => {
+                            let selectedPipelines = this.state.pipelineList.filter(
+                                (pipeline) => pipeline.checkbox.isChecked,
+                            )
+                            let newPipelines = response.result || []
+                            this.setState({
+                                view: ViewType.FORM,
+                                pipelineList: selectedPipelines.concat(newPipelines),
+                                openSelectPipeline: false,
+                            })
+                        })
+                        .catch((error) => {
+                            showError(error)
+                        })
+                })
             }
-        }
-        else if (theKeyCode === "Backspace") {
-            let unsavedFilterIndex = this.state.appliedFilters.findIndex(e => e.type && !e.value);
+        } else if (theKeyCode === 'Backspace') {
+            let unsavedFilterIndex = this.state.appliedFilters.findIndex((e) => e.type && !e.value)
             if (this.state.filterInput.length === 0 && unsavedFilterIndex >= 0) {
-                let state = { ...this.state };
-                state.appliedFilters.pop();
-                this.setState(state);
+                let state = { ...this.state }
+                state.appliedFilters.pop()
+                this.setState(state)
             }
         }
     }
 
-    showSesDropdown(): boolean {
-        let allEmails = this.state.selectedChannels?.filter(p => p.data.dest === "" || p.data.dest === "ses");
-        let show = allEmails.reduce((isValid, item) => {
-            isValid = isValid || validateEmail(item.data.recipient);
-            return isValid
-        }, false)
-        return show;
+    showEmailAgents(): boolean {
+        let allEmails = this.state.selectedChannels?.filter(
+            (p) =>
+                (p.data.dest === '' || p.data.dest === 'ses' || p.data.dest === 'smtp') &&
+                validateEmail(p.data.recipient),
+        )
+        return allEmails.length > 0
     }
 
     toggleSelectPipeline() {
-        let appliedFilters = this.state.appliedFilters;
+        let appliedFilters = this.state.appliedFilters
```

Comment:
Unnecessary initialization. Can remove this

Reply:
done